internal/resource/registry.go

```diff
@@ -68,14 +68,14 @@ type Registration struct {
 	Scope Scope
 }
 
-var ErrNeedData = errors.New("authorization check requires resource data")
+var ErrNeedResource = errors.New("authorization check requires the entire resource")
```

Comment:
Renamed this to be more generic because for `HealthStatus` I need to look at the `Owner` field to do authz